# Numerical tables for (insert name of paper here)

This notebook provides a searchable display for the tables of numerical values found in (paper). The first code cell below loads some Python packages needed to process the data.

In [2]:
import numpy as np
import pandas as pd
from pandas import DataFrame
import csv
from ipywidgets import widgets
import IPython

The next cell reads in the data from a collection CSV files generated (put information about how, who, etc. here).

The file names entered below will generate tables populated with $\LaTeX$-formatted numerical data. This will look nice, but you won't be able to work with the numbers. If you prefer to load numeric data (without formatting), append `_numeric` to the filename. (For example, replace `table9.csv` with `table9_numeric.csv`.)

In [3]:
table9 = pd.read_csv("table9.csv")
table10 = pd.read_csv("table10.csv",names=['$b_j$','$B_{j,1}$','$B_{j,2}$','$B_{j,3}$','$B_{j,4}$','$B_{j,5}$'])
table11 = pd.read_csv("table11.csv",names=['$b_j$','$B_{j,1}$','$B_{j,2}$','$B_{j,3}$','$B_{j,4}$','$B_{j,5}$'])
table13 = pd.read_csv("table13.csv",names=['$a$','$b$','$\mathcal{D}_0(a,b)$','$\mathcal{D}_1(a,b)$','$\mathcal{D}_2(a,b)$','$\mathcal{D}_3(a,b)$','$\mathcal{D}_4(a,b)$','$\mathcal{D}_5(a,b)$',])
table14 = pd.read_csv("table14.csv")
table15 = pd.read_csv("table15.csv")

In [4]:
table9

$b$              $\varepsilon(b)$
0           $20$    $ 4.26760 \cdot 10^{-5} $ 
1           $21$    $ 2.58843 \cdot 10^{-5} $ 
2           $22$    $ 1.56996 \cdot 10^{-5} $ 
3           $23$    $ 9.52229 \cdot 10^{-6} $ 
4           $24$    $ 5.77556 \cdot 10^{-6} $ 
5           $25$    $ 3.50306 \cdot 10^{-6} $ 
6           $26$    $ 2.12471 \cdot 10^{-6} $ 
7           $27$    $ 1.28871 \cdot 10^{-6} $ 
8           $28$    $ 7.81637 \cdot 10^{-7} $ 
9           $29$    $ 4.74087 \cdot 10^{-7} $ 
10          $30$     $2.87549 \cdot 10^{-7} $ 
11          $31$     $1.74407 \cdot 10^{-7} $ 
12          $32$     $1.05784 \cdot 10^{-7} $ 
13          $33$     $6.41607 \cdot 10^{-8} $ 
14          $34$     $3.89155 \cdot 10^{-8} $ 
15   $15\log 10$       $2.97255 \cdot 10^{-8}$
16          $35$     $2.36034 \cdot 10^{-8} $ 
17          $36$     $1.93378 \cdot 10^{-8} $ 
18          $37$     $1.93378 \cdot 10^{-8} $ 
19          $38$     $1.93378 \cdot 10^{-8} $ 
20          $39$     $1.93378 \cdot 10^{-8} $ 
21          $40$     $1.93378 \cdot 10^{-8} $ 
22          $41$     $1.93378 \cdot 10^{-8} $ 
23          $42$     $1.93378 \cdot 10^{-8} $ 
24          $43$     $1.93378 \cdot 10^{-8} $ 
25   $19\log 10$    $1.93378 \cdot 10 ^{-8} $ 
26          $44$   $ 1.72001 \cdot 10 ^{-8} $ 
27          $45$   $ 1.09073 \cdot 10 ^{-8} $ 
28          $46$   $ 6.94507 \cdot 10 ^{-9} $ 
29          $47$   $ 4.40123 \cdot 10 ^{-9} $ 
..            ...                          ...
225       $7200$    $5.25734 \cdot 10^{-17} $ 
226       $7300$    $4.26793 \cdot 10^{-17} $ 
227       $7400$    $3.46489 \cdot 10^{-17} $ 
228       $7500$    $2.81308 \cdot 10^{-17} $ 
229       $7600$    $2.28401 \cdot 10^{-17} $ 
230       $7700$    $1.85454 \cdot 10^{-17} $ 
231       $7800$    $1.50590 \cdot 10^{-17} $ 
232       $7900$    $1.22287 \cdot 10^{-17} $ 
233       $8000$    $9.93106 \cdot 10^{-18} $ 
234       $8100$    $8.06556 \cdot 10^{-18} $ 
235       $8200$    $6.55082 \cdot 10^{-18} $ 
236       $8300$    $5.32082 \cdot 10^{-18} $ 
237       $8400$    $4.32211 \cdot 10^{-18} $ 
238       $8500$    $3.51110 \cdot 10^{-18} $ 
239       $8600$    $2.85248 \cdot 10^{-18} $ 
240       $8700$    $2.31756 \cdot 10^{-18} $ 
241       $8800$    $1.88311 \cdot 10^{-18} $ 
242       $8900$    $1.53021 \cdot 10^{-18} $ 
243       $9000$    $1.24356 \cdot 10^{-18} $ 
244       $9100$    $1.01069 \cdot 10^{-18} $ 
245       $9200$    $8.21507 \cdot 10^{-19} $ 
246       $9300$    $6.67779 \cdot 10^{-19} $ 
247       $9400$    $5.42879 \cdot 10^{-19} $ 
248       $9500$    $4.41385 \cdot 10^{-19} $ 
249       $9600$    $3.58903 \cdot 10^{-19} $ 
250       $9700$    $2.91866 \cdot 10^{-19} $ 
251       $9800$    $2.37378 \cdot 10^{-19} $ 
252       $9900$    $1.93085 \cdot 10^{-19} $ 
253      $10000$    $1.57077 \cdot 10^{-19} $ 
254      $13900$    $6.03125 \cdot 10^{-23} $ 

[255 rows x 2 columns]

### Table 9: Sharper bounds for $\psi(x)$ - $|\psi(x) - x| < \varepsilon(b) x$ where $\varepsilon(b)$ is computable for $x \ge e^b$ fixed

Combined table of sharpest bounds for $|\psi(x)-x|<\varepsilon x$. Computed using Wedeniwski's partial verification of the Riemann Hypothesis<br> ($H_0=2\,445\,999\,556\,030$ `[51]`). For all $x\ge x_0=e^{b}$ we have $|\psi(x)-x|<\varepsilon(b) x$. 

In Table 9, values for $b=20,\ldots, 43$ are computed as in `[7, Theorem 2]` and `[6, Theorem 1]`. Values for $b=19\log 10,  \ldots, 2300$ are computed as in `[6, Theorem 1]`. Values for $b=2325,\ldots, 13900$ are computed as in Theorem 3.

In [ ]:
table9.style.set_properties(**{'width': '120px'});

In [ ]:
table9

### Table 10: Sharper bounds for $\theta(x)$: $x$ in a middle range ($e^{20} \le x \le e^{13900}$) 

Values for $ B_{j,k}$ in $|\theta(x)-x|<\frac{ B_{j,k}x}{(\log x)^k}$ calculated using the method described in Corollary 14.1, 
Section 4.2.

Each $ B_{j,k}$ is 
valid for $e^{b_j}\le x \le e^{b_{j+1}}$. The last line is valid for $e^{10000} \le x \le e^{13900}$.

In [ ]:
table10

### Table 11: Sharper bounds for $\theta(x)$: $x$ in a middle range ($e^{20} \le x \le e^{13900}$)

Supplement for Table 10. All $\mathcal{B}_{j,k}$ values are valid for $x \in [e^{b_j},e^{13900}]$.

### Table 12: Lower bound for first values of $x \in [e^{J_0},10^{19}]$

Values of $\mathcal{C}_{b,k}$ in $\theta(x) > x - \frac{\mathcal{C}_{b,k} x}{(\log x)^k}$ calculated using the method described in Corollary 15.1, Section 4.4.
Each $\mathcal{C}_{b,k}$ is valid for $e^b \le x \le 10^{19} \simeq e^{43.749}$.

$$\begin{array}{cccccccc} 
 & b & \mathcal{C}_{b,1} & \mathcal{C}_{b,2} & \mathcal{C}_{b,3} & \mathcal{C}_{b,4} & \mathcal{C}_{b,5} & \\ 
 \hline
 &&&&&&& \\[-1em]
\rlap{\text{Calculated using } c=-0.8, C=0.81, \text{ each value valid up to } 5\cdot10^{10}\simeq e^{24.635}.}\\ \hline
&&&&&&& \\[-1em]
&  20  &  1.68440 \cdot 10^{-3}  &  3.36880 \cdot 10^{-2}  &  6.73750 \cdot 10^{-1}  &  1.34750 \cdot 10^{1}  &  2.69500 \cdot 10^{2 }  & \\
&  21  &  1.06840 \cdot 10^{-3}  &  2.24350 \cdot 10^{-2}  &  4.71140 \cdot 10^{-1}  &  9.89390 \cdot 10^{0}  &  2.07780 \cdot 10^{2 }  & \\
&  22  &  6.76540 \cdot 10^{-4}  &  1.48840 \cdot 10^{-2}  &  3.27450 \cdot 10^{-1}  &  7.20380 \cdot 10^{0}  &  1.58490 \cdot 10^{2 }  & \\
&  23  &  4.27800 \cdot 10^{-4}  &  9.83920 \cdot 10^{-3}  &  2.26310 \cdot 10^{-1}  &  5.20500 \cdot 10^{0}  &  1.19720 \cdot 10^{2 }  & \\
&  24  &  2.70120 \cdot 10^{-4}  &  6.48290 \cdot 10^{-3}  &  1.55590 \cdot 10^{-1}  &  3.73410 \cdot 10^{0}  &  8.96190 \cdot 10^{1 }  & \\
\hline
&&&&&&& \\[-1em]
\rlap{\text{Calculated using } c=-0.88, C=0.86, \text{ each value valid up to } 32\cdot10^{12}\simeq e^{31.097}.}\\ \hline
&&&&&&& \\[-1em]
&  \log(5\cdot 10^{10})  &  2.01560 \cdot 10^{-4}  &  4.96540 \cdot 10^{-3}  &  1.22330 \cdot 10^{-1}  &  3.01350 \cdot 10^{0}  &  7.42380 \cdot 10^{1 }  & \\
&  25  &  1.70330 \cdot 10^{-4} & 4.25830 \cdot 10^{-3}  &  1.06460 \cdot 10^{-1}  &  2.66140 \cdot 10^{0}  &  6.65350 \cdot 10^{1 }  & \\
&  26  &  1.10220 \cdot 10^{-4}  &  2.86560 \cdot 10^{-3}  &  7.45050 \cdot 10^{-2}  &  1.93720 \cdot 10^{0}  &  5.03650 \cdot 10^{1 }  & \\
&  27  &  6.93270 \cdot 10^{-5}  &  1.87190 \cdot 10^{-3}  &  5.05400 \cdot 10^{-2}  &  1.36460 \cdot 10^{0}  &  3.68430 \cdot 10^{1 }  & \\
&  28  &  4.35580 \cdot 10^{-5}  &  1.21970 \cdot 10^{-3}  &  3.41500 \cdot 10^{-2}  &  9.56180 \cdot 10^{-1}  &  2.67730 \cdot 10^{1 }  & \\
&  29  &  2.73380 \cdot 10^{-5}  &  7.92780 \cdot 10^{-4}  &  2.29910 \cdot 10^{-2}  &  6.66730 \cdot 10^{-1}  &  1.93360 \cdot 10^{1 }  & \\
&  30  &  1.71400 \cdot 10^{-5}  &  5.14180 \cdot 10^{-4}  &  1.54260 \cdot 10^{-2}  &  4.62760 \cdot 10^{-1}  &  1.38830 \cdot 10^{1 }  & \\
&  31  &  1.07350 \cdot 10^{-5}  &  3.32790 \cdot 10^{-4}  &  1.03170 \cdot 10^{-2}  &  3.19810 \cdot 10^{-1}  &  9.91400 \cdot 10^{0 }  & \\
\hline
&&&&&&& \\[-1em]
\rlap{\text{Calculated using } -c=C=0.94, \text{ each value valid up to } 10^{19}\simeq e^{43.749}.}\\ \hline
&&&&&&& \\[-1em]
&  \log(3.2\cdot10^{13})  &  1.02600 \cdot 10^{-5}  &  3.19040 \cdot 10^{-4}  &  9.92090 \cdot 10^{-3}  &  3.08510 \cdot 10^{-1}  &  9.59360 \cdot 10^{0 }  & \\
&  32  &  6.71750 \cdot 10^{-6}  &  2.14960 \cdot 10^{-4}  &  6.87870 \cdot 10^{-3}  &  2.20120 \cdot 10^{-1}  &  7.04380 \cdot 10^{0 }  & \\
&  33  &  4.38000 \cdot 10^{-6}  &  1.44540 \cdot 10^{-4}  &  4.76990 \cdot 10^{-3}  &  1.57410 \cdot 10^{-1}  &  5.19440 \cdot 10^{0 }  & \\
&  34  &  2.73610 \cdot 10^{-6}  &  9.30270 \cdot 10^{-5}  &  3.16300 \cdot 10^{-3}  &  1.07540 \cdot 10^{-1}  &  3.65640 \cdot 10^{0 }  & \\
&  35  &  1.70780 \cdot 10^{-6}  &  5.97730 \cdot 10^{-5}  &  2.09210 \cdot 10^{-3}  &  7.32220 \cdot 10^{-2}  &  2.56280 \cdot 10^{0 }  & \\
&  36  &  1.06520 \cdot 10^{-6}  &  3.83460 \cdot 10^{-5}  &  1.38050 \cdot 10^{-3}  &  4.96960 \cdot 10^{-2}  &  1.78910 \cdot 10^{0 }  & \\
&  37  &  6.63850 \cdot 10^{-7}  &  2.45630 \cdot 10^{-5}  &  9.08810 \cdot 10^{-4}  &  3.36260 \cdot 10^{-2}  &  1.24420 \cdot 10^{0 }  & \\
&  38  &  4.13450 \cdot 10^{-7}  &  1.57120 \cdot 10^{-5}  &  5.97020 \cdot 10^{-4}  &  2.26870 \cdot 10^{-2}  &  8.62100 \cdot 10^{-1 }  & \\
&  39  &  2.57330 \cdot 10^{-7}  &  1.00360 \cdot 10^{-5}  &  3.91400 \cdot 10^{-4}  &  1.52650 \cdot 10^{-2}  &  5.95320 \cdot 10^{-1 }  & \\
&  40  &  1.60060 \cdot 10^{-7}  &  6.40240 \cdot 10^{-6}  &  2.56100 \cdot 10^{-4}  &  1.02440 \cdot 10^{-2}  &  4.09750 \cdot 10^{-1 }  & \\
&  41  &  9.94970 \cdot 10^{-8}  &  4.07940 \cdot 10^{-6}  &  1.67260 \cdot 10^{-4}  &  6.85740 \cdot 10^{-3}  &  2.81160 \cdot 10^{-1 }  & \\
&  42  &  6.18140 \cdot 10^{-8}  &  2.59620 \cdot 10^{-6}  &  1.09040 \cdot 10^{-4}  &  4.57970 \cdot 10^{-3}  &  1.92350 \cdot 10^{-1 }  & \\
&  43  &  3.83820 \cdot 10^{-8}  &  1.65050 \cdot 10^{-6}  &  7.09680 \cdot 10^{-5}  &  3.05170 \cdot 10^{-3}  &  1.31220 \cdot 10^{-1 }  & \\
\hline
\end{array} $$

### Table 13: Lower bound for $x \le 10^{19}$ (Numerical Verification)

Values for $\mathcal{D}_k(a,b)$ in 
\\
 $\mathcal{D}_k(a,b) = \max_{n_0 \le n \le n_1} \left( \frac{(\log p_{n})^k \cdot (p_{n} - \theta(p_{n-1}))}{p_{n}} \right)$
\\
calculated using the method described in Lemma 16, Section 4.5. We give values for selected ranges between $1$ and $7.0 \cdot 10^{11}$.

### Table 14: Values for Theorem 2 for $k = 0$.

Values of $m_0$ in $\theta(x) \ge x(1-m_0)$ and values of $M_0$ in $\theta(x) \le x(1+M_0)$ from Theorem 2,
using the methods shown in Lemmas 18 and 17 respectively. Each value is valid for $x \ge \log X_0 = \log X_1$.

### Table 15: Values for Theorem 2 for $k \in \{1,2,3,4,5\}$.

Values of $m_k = M_k$ in $ |\theta(x) - x| < \frac{m_k x}{(\log x)^k}$ as in Theorem 2. 
Combined theoretical results from Table 8, 11, and 12. 
Each value is valid for $x \ge X_0 = X_1$. Note that for $x \le 10^{19}$, we have $M_k = 0$, and for $x \le e^{27}$, we have numerical results 
in Table 13.